File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        vector<int> result(2);
        map<int, int> m;
        for (auto it : nums)
            m[it]++;
        for (int i = 1; i <= nums.size(); i++)
        {
            if (m[i] == 0)
                result[1] = i;
            else if (m[i] == 2)
                result[0] = i;
        }
        
        return (result);
    }
};

File: 0645-set-mismatch/0645-set-mismatch.py
class Solution(object):
    def findErrorNums(self, nums):
        numFrequency = {};
        result = [0 for _ in range(2)]
        for num in nums:
            numFrequency[num] = numFrequency.get(num, 0) + 1;
        for i in range(1, len(nums) + 1):
            if numFrequency.get(i) == None:
                result[1] = i;
            if numFrequency.get(i) > 1:
                result[0] = i;
        return (result);